protocol/xmrtaker/instance.go

```diff
@@ -64,12 +64,17 @@ func NewInstance(cfg *Config) (*Instance, error) {
 }
 
 func (inst *Instance) checkForOngoingSwaps() error {
-	swaps, err := inst.backend.SwapManager().GetOngoingSwaps()
+	ongoingIDs, err := inst.backend.SwapManager().GetOngoingSwapOfferIDs()
 	if err != nil {
 		return err
 	}
 
-	for _, s := range swaps {
+	for _, offerID := range ongoingIDs {
+		s, err := inst.backend.SwapManager().GetOngoingSwap(*offerID)
+		if err != nil {
+			return err
+		}
```

Comment:
why is this changed? can you add a function to get all ongoing swaps like before?

Reply:
`GetOngoingSwapsSnapshot` returns all the swaps (as a deep copy) and it is still there. I thought the code was easier to audit if we just got the IDs here and later pull the non-deep-copies on an individual bases when restarting a swap.

Reply:
I don't think it makes a difference and this results in 2 iterations instead of 1, and we're iterating through all of them either way

Reply:
maybe implement a `GetOngoingSwaps` that returns all the swaps not as a copy?

Reply:
I downgraded the code from getting the full, modifiable object list here, because there is nothing at this high level in the code that needs access to anything but the offer IDs. By getting the modifiable copies lower down on the stack on an individual basis, it shortens the chain-of-custody for the modifiable objects and reduces the API surface area that returns modifiable objects. Auditing synchronous access to objects is hard and I was hoping these changes make the code easier to audit for correct use of locks.